torch/csrc/jit/passes/peephole_list_idioms.cpp

```diff
@@ -23,13 +27,36 @@ c10::optional<size_t> normalizeIndex(int64_t index, size_t len) {
   }
 }
 
+bool isListOrDict(Value* v) {
+  return v->type()->cast<ListType>() || v->type()->cast<DictType>();
+}
+
+struct DictNode {
+  explicit DictNode(Node* dict_creation_node) {
+    for (size_t i = 0; i < dict_creation_node->inputs().size(); i += 2) {
+      Value* key = dict_creation_node->input(i);
+      bool is_constant = toIValue(key) != c10::nullopt;
+      if (is_constant && dict.find(key) == dict.end()) {
```

Comment:
we dont want to look on `Value *` to check on overlap we want to check on overlap of the IValue value `key`. i would just handle common immutable keys for now (int, str, float) https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/jit_type.h#L761